src/graphnet/deployment/i3modules/deployer.py

```diff
@@ -0,0 +1,204 @@
+    from I3Tray import I3Tray
+
+if has_torch_package or TYPE_CHECKING:
+    import torch
+
+
+class GraphNeTI3Deployer:
+    """Deploys graphnet i3 modules to i3 files.
+
+    Modules are applied in the order in which they appear in graphnet_modules.
+    """
+
+    def __init__(
+        self,
+        graphnet_modules: Union[
+            GraphNeTI3Module,
+            I3InferenceModule,
+            I3PulseCleanerModule,
+            List[GraphNeTI3Module],
+            List[I3InferenceModule],
+            List[I3PulseCleanerModule],
+            List[
+                Union[
+                    GraphNeTI3Module,
+                    I3InferenceModule,
+                    I3PulseCleanerModule,
+                ]
+            ],
+        ],
+        gcd_file: str,
+        n_workers: int = 1,
+    ) -> None:
+        """Initialize the deployer.
+
+        Will apply graphnet i3 modules to i3 files in the order in which they appear in graphnet_modules.
+        Each module is run independently.
+
+        Args:
+            graphnet_modules: List of graphnet i3 modules. Order of appearence in the list determines order of deployment.
+            gcd_file: path to gcd file.
+            n_workers: Number of workers. The deployer will divide the number of input files across workers. Defaults to 1.
+        """
+        # This makes sure that one worker cannot access more than 1 core's worth of compute.
+        torch.set_num_threads(1)
+        torch.set_num_interop_threads(1)
+        # Check
+        if isinstance(graphnet_modules, list):
+            self._modules = graphnet_modules
+        else:
+            self._modules = [graphnet_modules]
+        self._gcd_file = gcd_file
+        self._n_workers = n_workers
+
+    def _setup_settings(
+        self, input_files: List[str], output_folder: str
+    ) -> List:
+        """Will construct the list of arguments for each worker."""
+        try:
+            os.makedirs(output_folder)
+        except FileExistsError:
+            assert (
+                1 == 2
+            ), f"{output_folder} already exists. To avoid overwriting existing files, the process has been stopped."
+        settings = []
+        if self._n_workers > len(input_files):
+            self._n_workers = len(input_files)
+
+        if self._n_workers > 1:
+            file_batches = np.array_split(input_files, self._n_workers)
+            for i in range(self._n_workers):
+                settings.append(
+                    [
+                        file_batches[i],
+                        self._gcd_file,
+                        output_folder,
+                        self._modules,
+                    ]
+                )
+        else:
+            settings = [
+                input_files,  # type: ignore
+                self._gcd_file,  # type: ignore
+                output_folder,  # type: ignore
+                self._modules,  # type: ignore
+            ]  # type: ignore
+        return settings
```

Comment:
I have always had a hard time with settings-as-lists, because they require you to keep track of what property a certain list index corresponds to. Could I suggest instead adding a small class at the top of this file,
```python
@dataclass
class Settings:
    i3_files: List[str]
    gcd_file: str
    output_folder: str
    modules: List[GraphNeTI3Module]
```
Then you can do stuff like `settings.i3_files` rather than `settings[ix]` or unwrapping like `(i3_files, gcd_file, ...) = settings` which is probably less maintainable.

Then you can use this class in this call as suggested below:

```suggestion
    def _prepare_settings(
        self, input_files: List[str], output_folder: str
    ) -> List[Settings]:
        """Will prepare the settings for each worker."""
        try:
            os.makedirs(output_folder)
        except FileExistsError:
            assert (
                False
            ), f"{output_folder} already exists. To avoid overwriting existing files, the process has been stopped."

        if self._n_workers > len(input_files):
            self._n_workers = len(input_files)

        if self._n_workers > 1:
            file_batches = np.array_split(input_files, self._n_workers)
            settings: List[Settings] = []
            for i in range(self._n_workers):
                settings.append(
                    Settings(
                        file_batches[i],
                        self._gcd_file,
                        output_folder,
                        self._modules,
                    )
                )
        else:
            settings = [Settings(
                input_files,
                self._gcd_file,
                output_folder,
                self._modules,
            )]

        return settings
```